WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderSelectInterestsViewController.swift

```diff
@@ -23,16 +23,20 @@ class ReaderSelectInterestsViewController: UIViewController {
     @IBOutlet weak var collectionView: UICollectionView!
     @IBOutlet weak var buttonContainerView: UIView!
     @IBOutlet weak var nextButton: FancyButton!
+    @IBOutlet weak var activityIndicatorView: UIActivityIndicatorView!
 
-    // MARK: - Mock Data Source
-    private let fakeDataSource = InterestsDataSource(fileName: "interests.json")
+    // MARK: - Data
+    private let dataSource: ReaderInterestsDataSource = ReaderInterestsDataSource()
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
+        dataSource.delegate = self
```

Comment:
🤖  DANGER 🤖  this is a retain cycle.
You should declare the delegate in the data source as weak but before you change it:
* Put a breakpoint in `BlogDetrailsViewController viewDidAppear` 
* Display select interests screen 
* Dismiss select interest screen
* Go to "My Site". Would hit breakpoint
* Tap on Debug Memory Graph icon and see the retain cycle that leaks the data source, view models and view controller
![Screen Shot 2020-07-08 at 5 54 35 PM](https://user-images.githubusercontent.com/1335657/86985228-a5653900-c145-11ea-9b08-6a83c5a7eeb7.png)

Reply:
D'oh.. 🤦‍♀️  Thanks for catching that!